SpringProject/src/it/java/com/gradproject2019/AuthControllerIT.java

```diff
@@ -0,0 +1,122 @@
+package com.gradproject2019;
+
+import com.gradproject2019.auth.data.LoginDto;
+import com.gradproject2019.auth.persistance.Token;
+import com.gradproject2019.auth.repository.AuthRepository;
+import com.gradproject2019.users.persistance.User;
+import com.gradproject2019.users.repository.UserRepository;
+import com.gradproject2019.utils.PasswordUtils;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.web.client.TestRestTemplate;
+import org.springframework.boot.web.server.LocalServerPort;
+import org.springframework.core.ParameterizedTypeReference;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.ResponseEntity;
+import org.springframework.http.client.HttpComponentsClientHttpRequestFactory;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+
+import static org.springframework.http.HttpMethod.POST;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest(webEnvironment= SpringBootTest.WebEnvironment.RANDOM_PORT)
+public class AuthControllerIT {
+
+    @Autowired
+    private AuthRepository authRepository;
+
+    @Autowired
+    private UserRepository userRepository;
+
+    @Autowired
+    private TestRestTemplate restTemplate;
+
+    @LocalServerPort
+    int testServerPort = 8080;
```

Comment:
You do not need to hardcode port here, because above you have used SpringBootTest.WebEnvironment.RANDOM_PORT

Reply:
How would we assign the port to the `baseUrl` later in the code, line 55 I think?